Dispose rejected requests

davidfowl opened the pull request:
- Rejected requests never get scheduled, so make sure they get disposed before getting the new request from the queue

davidfowl committed:
Dispose rejected requests
- Rejected requests never get scheduled, so make sure they get disposed before getting the new request from the queue

davidfowl commented on src/Servers/HttpSys/src/MessagePump.cs:
```diff
@@ -192,6 +192,9 @@ private async Task ProcessRequestsWorker()
                     if (!Listener.ValidateRequest(requestContext))
                     {
+                        // Dispose the request
+                        requestContext.Dispose();
```
This logic *could* be moved to ValidateRequest but it felt wrong to do it there. I'm fine either way..

Tratcher replied:
Should it also be disposed in the catch block in case ValidateRequest throws?

davidfowl replied:
Good catch

davidfowl committed:
Handle request validation throwing
- We also want to dispose if request validation fails
- Log an error if it throws
